### ISSUE TITLE
[Feature Request] Resize podcast cover art to reduce disk space

### ISSUE THREAD

**Issue by neodyne** [gpodder/gpodder#1072]
After I add a new podcast to gPodder, I usually run an external powershell script from the config directory:

```
#!/usr/bin/env pwsh
Add-Type -AssemblyName System.Drawing # requires libgdiplus with pwsh on linux, ffmpeg in path
ForEach ($cover in (gci -File Downloads/folder.* -Exclude *.gif -Recurse).FullName){
	$img=New-Object System.Drawing.Bitmap $cover
	if ($img.Height -gt 300 -or $img.Width -gt 300){
		$img.Dispose()
		ffmpeg -y -i $cover -vf scale=w=300:h=300:force_original_aspect_ratio=decrease $cover.replace("folder.","folder_small.") 2>$null
		ri -LiteralPath $cover -Force
		rni -LiteralPath $cover.replace("folder.","folder_small.") $cover -Force
	}
	else {$img.Dispose()}
}
```

This is done because with 100+ podcast subscriptions, gPodder takes quite a while to load, if it has to parse all the covers at their original resolution (for some reason some podcasts decide to have a 3000x3000 30+ MB png as their cover art). If I understand correctly, the cover art is (only?) used in the GUI in various places, where it is already fairly small (in the list on the left side, and podcast settings).

Is it possible for gPodder to either:

- shrink the cover art to an arbitrary resolution upon subscription to a podcast, if the cover art is bigger
(similar to what the above snippet does, but probably not what everyone would want)
- generate a smaller "thumbnail" cover art for podcasts (with resolution as a variable), which will then get used in the GUI to speed up the startup
- or maybe include an extension that does something similar (likely depending on ffmpeg). However, I am not sure how the interaction with either refreshing the image, or the tagging extension would work.

My python knowledge could use some work, but if you think having this as an extension would be meaningful, I could, with some input, try to write one.

**Comment by auouymous** [comment]
Good idea! I have 76 channels and only a few are barely over 100kB, but it would be nice if each thumbnail were 100 times smaller, which might decrease the startup time and memory usage. I'd shrink them to the icon size since I don't care what the image looks like in channel dialog.

There don't appear to be any hooks for an extension to be notified when coverart is downloaded. Gpodder would either need to do this or a hook added. And a setting would be needed since users with high DPI monitors might want larger images.

There are a couple issues to solve.

The setting should probably resize all images when changed. This requires a UI and a button to perform the resize, because it can't resize all images as you adjust the setting. I'm not sure if extensions can add tabs to the preferences dialog.

Increasing the setting would require re-downloading all images. This could be indicated on the UI and the resize button would take longer when size is increased.

The biggest issue is how to handle sizes larger than the downloaded image. The feature would need to fetch the image's current size and not resize to a larger size than the download image (waste of space). But it might also need to save something indicating the image is the download image and not try to re-download it.

**Comment by auouymous** [comment]
This is still needed to reduce disk space for those with channels that have 3000x3000 images and don't want to disable thumbnails.
